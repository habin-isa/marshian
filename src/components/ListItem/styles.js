import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from {
    opacity: 0;
  }

  to {
    opacity: 1;
  }
`;

export const Wrapper = styled.div`
	display: flex;
	flex-direction: column;
	justify-content: space-evenly;
	align-items: center;
	// background-color: #433756;
	margin: 1em;
	// width: 9em;
	// height: 9em;
	border-radius: 20px;
	color: white;
	// padding: 4em;
	// visibility: visible;
	// animation: ${fadeIn} 500ms ease-in;
  // transition: visibility 500ms ease-in;
  letter-spacing: 1px;
  position: relative;
`;

export const TopContainer = styled.div`
  display: flex;
  flex-direction: column;
  background-color: #f8f9fa;
  margin: 10px;
  height: 140px;
  width: 350px;
  @media screen and (max-width: 416px) {
    width: 300px;
  }
  padding: 1.5em 2em;
  justify-content: space-between;
  &:hover {
    opacity: 0.9;
    cursor: pointer;
  }
`;

export const Container = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;

export const VesselImg = styled.img`
  width: 45px;
  height: 45px;
  border-radius: 50px;
`;

export const TitleContainer = styled.div`
  display: flex;
  flex-direction: column;
`;

export const Title = styled.div`
  font-weight: 500;
  font-size: 18px;
  color: #4f4f4f;
  margin-bottom: 2px;
`;

export const Icon = styled.img`
  width: 35px;
  height: 35px;
`;

export const Subtitle = styled.span`
  text-transform: capitalize;
  font-weight: 100;
  font-size: 12px;
  color: #828282;
`;

export const TagsContainer = styled.div`
  display: flex;
  align-items: center;
  width: fit-content;
`;

export const TagIcon = styled.img`
  width: 20px;
  height: 20px;
  margin-right: 5px;
`;

export const Tag = styled.div`
  background-color: white;
  height: 35px;
  text-align: center;
  border-radius: 20px;
  border: 2px solid #f2f2f2;
  color: #828282;
  font-weight: 400;
  font-size: 14px;
  display: flex;
  display: flex;
  align-items: center;
  min-width: 45%;
  width: fit-content;
  justify-content: center;
  padding: 2px 10px;
  font-size: 12px;
  margin: 10px;
`;

export const DetailsContainer = styled.div`
  background: white;
  position: absolute;
  min-height: 130px;
  width: 350px;
  padding: 2em;
  top: 200px;
  z-index: 999;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const LinkContainer = styled.a`
  display: flex;
  align-items: center;
  justify-content: space-evenly;
  text-decoration: none;
  height: 35px;
  text-align: center;
  border-radius: 20px;
  border: 2px solid #f2f2f2;
  color: #828282;
  font-weight: 400;
  font-size: 14px;
  display: flex;
  align-items: center;
  min-width: 45%;
  width: fit-content;
  justify-content: center;
  padding: 2px 10px;
  font-size: 12px;
  margin: 10px;
  &:hover {
    opacity: 0.8;
  }
`;
