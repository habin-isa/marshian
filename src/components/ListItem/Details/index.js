import React from 'react';
import * as S from './styles';

import { object, bool } from 'prop-types';

const Details = ({ vessel, dragonsActive }) => {
  return (
    <div>
      <S.Description>
        {vessel.description}
        {vessel.active === true ? ' (Active)' : ' (Inactive)'}
      </S.Description>
      <S.Image src={vessel.flickr_images[0]} alt={vessel.id} />
      <S.Image src={vessel.flickr_images[1]} alt={vessel.id} />
      <S.CardContainer>
        {dragonsActive === true ? (
          <S.Card>
            <S.Figure>1</S.Figure>
            Heat Shield
            <S.Figure>{vessel.heat_shield.size_meters}</S.Figure>
            /m Size
            <S.Figure>{vessel.heat_shield.temp_degrees}</S.Figure>
            /degrees Temp
          </S.Card>
        ) : (
          <S.Card>
            <S.Figure>{vessel.boosters}</S.Figure>
            Boosters
            <S.Figure>{vessel.mass.kg}</S.Figure>
            /kg Mass
            <S.Figure>{vessel.stages}</S.Figure>
            Stages
          </S.Card>
        )}
        {dragonsActive === true ? (
          <S.Card>
            <S.Figure>{vessel.launch_payload_mass.kg}</S.Figure>
            /kg Launch mass
            <S.Figure>{vessel.orbit_duration_yr}</S.Figure>
            /yrs Orbit Duration
            <S.Figure>{vessel.sidewall_angle_deg}</S.Figure>
            /degrees Sidewall
          </S.Card>
        ) : (
          <S.Card>
            <S.Figure>1</S.Figure>
            First stage
            <S.Figure>{vessel.first_stage.burn_time_sec}</S.Figure>
            /sec Burn time
            <S.Figure>{vessel.first_stage.fuel_amount_tons}</S.Figure>
            /tons Fuel amount
          </S.Card>
        )}
      </S.CardContainer>
    </div>
  );
};

export default Details;

Details.propTypes = {
  vessel: object,
  dragonsActive: bool.isRequired
};

Details.defaultProps = {
  vessel: {}
};
