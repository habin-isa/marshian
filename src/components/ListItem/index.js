import React, { useState } from 'react';
import * as S from './styles';

import Details from './Details';
import ActiveIcon from '../../assets/active.png';
import InactiveIcon from '../../assets/inactive.png';
import CapsuleIcon from '../../assets/capsule.png';
import CrewIcon from '../../assets/crew.png';
import PercentageIcon from '../../assets/percentage.png';
import MoneyIcon from '../../assets/money.png';
import emptyLikeIcon from '../../assets/empty-like.svg';
import fullLikeIcon from '../../assets/full-like.svg';
import { object, bool } from 'prop-types';

const ListItem = ({ vessel, dragonsActive, handleLikeCount, handleLikeTitle }) => {
  const [showDetails, setShowDetails] = useState(false);
  const [itemLiked, setItemLiked] = useState(false);

  const handleLikeClick = () => {
    setItemLiked(!itemLiked);
    if (itemLiked) {
      handleLikeCount(-1);
    } else {
      handleLikeCount(1);
    }
  };
  return (
    <S.Wrapper>
      <S.TopContainer>
        {itemLiked ? (
          <S.HeartIcon
            src={fullLikeIcon}
            onClick={() => {
              handleLikeClick();
              handleLikeTitle('', vessel.id);
            }}
          />
        ) : (
          <S.HeartIcon
            src={emptyLikeIcon}
            onClick={() => {
              handleLikeClick();
              handleLikeTitle(vessel.name || vessel.rocket_name, vessel.id);
            }}
          />
        )}
        <S.Container onClick={() => setShowDetails(!showDetails)}>
          <S.TitleContainer>
            <S.VesselImg src={vessel.flickr_images[0]} alt={vessel.id} />
            <S.Text>
              <S.Title>{vessel.name || vessel.rocket_name}</S.Title>
              <S.Subtitle>First flight: {vessel.first_flight}</S.Subtitle>
            </S.Text>
          </S.TitleContainer>
          <S.Icon src={vessel.active ? ActiveIcon : InactiveIcon} />
        </S.Container>
        <S.TagsContainer>
          {dragonsActive === true ? (
            <S.Tag>
              <S.TagIcon src={CapsuleIcon} />
              Capsule
            </S.Tag>
          ) : (
            <S.Tag>
              <S.TagIcon src={PercentageIcon} />
              Success rate: {vessel.success_rate_pct}%
            </S.Tag>
          )}
          {dragonsActive === true ? (
            <S.Tag>
              <S.TagIcon src={CrewIcon} />
              Crew capacity: {vessel.crew_capacity}
            </S.Tag>
          ) : (
            <S.Tag>
              <S.TagIcon src={MoneyIcon} />${vessel.cost_per_launch}
            </S.Tag>
          )}
          {showDetails === false ? (
            <S.Caret onClick={() => setShowDetails(true)}>▼</S.Caret>
          ) : (
            <S.Caret onClick={() => setShowDetails(false)}>▲</S.Caret>
          )}
        </S.TagsContainer>
      </S.TopContainer>
      {showDetails === true && (
        <div>
          {' '}
          <Details vessel={vessel} dragonsActive={dragonsActive} handleCaret={() => setShowDetails(!showDetails)} />
        </div>
      )}
    </S.Wrapper>
  );
};

export default ListItem;

ListItem.propTypes = {
  vessel: object,
  dragonsActive: bool
};

ListItem.defaultProps = {
  vessel: {},
  dragonsActive: true
};
