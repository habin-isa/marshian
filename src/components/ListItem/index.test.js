import React from 'react';
import { shallow } from 'enzyme';
import { render } from '@testing-library/react';
import ListItem from '.';

describe('ListItem component', () => {
  const props = {
    vessel: {
      name: 'Dragon 1',
      active: true,
      flickr_images: ['https://i.imgur.com/9fWdwNv.jpg'],
      first_flight: '2006-08-17',
      success_rate_pct: 80,
      crew_capacity: 0
    },
    dragonsActive: true
  };
  it('renders with correct data', () => {
    const { container } = render(<ListItem vessel={props.vessel} dragonsActive={props.dragonsActive} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
