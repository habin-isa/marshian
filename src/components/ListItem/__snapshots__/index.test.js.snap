// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListItem component renders with correct data 1`] = `
<div
  class="sc-bdnylx mIRxC"
>
  <div
    class="sc-gtssRu cXtQAK"
  >
    <div
      class="sc-dlnjPT bHESxQ"
    >
      <img
        class="sc-hKFyIo hmcCbS"
        src="https://i.imgur.com/9fWdwNv.jpg"
      />
      <div
        class="sc-eCApGN eYskGV"
      >
        <div
          class="sc-jSFkmK gMinZq"
        >
          Dragon 1
        </div>
        <span
          class="sc-iCoHVE kJKVjK"
        >
          First flight: 
          2006-08-17
        </span>
      </div>
      <img
        class="sc-gKAblj sfmgn"
        src="active.png"
      />
    </div>
    <div
      class="sc-fujyUd jnFYcx"
    >
      <div
        class="sc-jrsJCI bcGbNu"
      >
        <img
          class="sc-pNWxx lQFdD"
          src="capsule.png"
        />
        Capsule
      </div>
      <div
        class="sc-jrsJCI bcGbNu"
      >
        <img
          class="sc-pNWxx lQFdD"
          src="crew.png"
        />
        Crew capacity: 
        0
      </div>
    </div>
  </div>
  <div />
</div>
`;
